cwltool/factory.py

```diff
@@ -28,34 +35,74 @@ def __init__(self, t, factory):  # type: (Process, Factory) -> None
 
     def __call__(self, **kwargs):
         # type: (**Any) -> Union[Text, Dict[Text, Text]]
-        runtime_context = self.factory.runtime_context.copy()
-        runtime_context.basedir = os.getcwd()
-        out, status = self.factory.executor(self.t, kwargs, runtime_context)
+        out, status = self.factory.executor(
+            self.t, kwargs, self.factory.runtimeContext)
         if status != "success":
             raise WorkflowStatus(out, status)
         else:
             return out
 
+
 class Factory(object):
     def __init__(self,
-                 executor=None,        # type: tCallable[...,Tuple[Dict[Text,Any], Text]]
-                 loading_context=None,  # type: LoadingContext
-                 runtime_context=None   # type: RuntimeContext
-                ):  # type: (...) -> None
+                 argsl=None,                   # type: List[str]
+                 args=None,                    # type: argparse.Namespace
+                 executor=None,                # type: Callable[..., Tuple[Dict[Text, Any], Text]]
```

Comment:
```suggestion
                 executor=None,                # type: tCallable[..., Tuple[Dict[Text, Any], Text]]
```
The `type` `Callable` was renamed to `tCallable` in this file to not clash with the `class` `Callable` that we defined also here.